web/src/main/webapp/WEB-INF/pages/application_state.jsp

```diff
@@ -51,27 +65,33 @@
               <h4 class="glyphicon glyphicon-edit pull-right" id="latestStateEditButton" onclick="editLatestStateOnClick()"></h4>
               </c:if>
           </div>
-          <form class="form form-horizontal" role="form" method="POST" action="#">
-
-              <input type="hidden" name="applicationId" value="${applicationId}" />
-              <input type="hidden" name="id" value="${data.id}"/>
-              <input type="hidden" name="stateId" value="${data.stateId}"/>
+          <form data-toggle="validator" class="form form-horizontal" role="form" method="POST" id="create-state-form" action="#">
+              <c:if test="${stat.first}">
+                <input type="hidden" name="id" value="${data.id}" data-bind="valueWithInit: 'id'"/>
+                <input type="hidden" name="stateId" value="${data.stateId}" data-bind="valueWithInit: 'stateId'"/>
+              </c:if>
 
               <div class="form-group col-sm-12 col-md-12 col-lg-12">
                   <label for="name" class="control-label col-sm-4"><spring:message code="statehistory.field.date"/></label>
                   <div class="col-sm-8">
                       <p class="form-control-static <c:if test="${stat.first}">stateData"</c:if> id="creationDateP">${data.creationDate}</p>
                       <c:if test="${stat.first}">
-                          <input class="stateInput" type="text" name="creationDate" id="creationDateInput" style="display:none" value="${data.creationDate}" data-formatter="creationDateFormatter">
+                          <input class="stateInput" type="text" name="creationDate" id="creationDateInput" style="display:none" value="${data.creationDate}"
```

Comment:
Do not inline styles into html markup as attributes, prefer to use classes.

In this case style="display:none" can be replaced with class="hidden", as hidden class is defined by bootstrap.